src/main/scala/com/wavesplatform/Coordinator.scala

```diff
@@ -24,7 +24,7 @@ import scala.util.{Left, Right}
 object Coordinator extends ScorexLogging with Instrumented {
   def processFork(checkpoint: CheckpointService, history: History, blockchainUpdater: BlockchainUpdater,
                   stateReader: StateReader, utxStorage: UtxPool, time: Time, settings: WavesSettings,
-                  blockchainReadiness: AtomicBoolean, featureProvider: FeatureProvider)
+                  blockchainReadiness: AtomicBoolean, featureProvider: FeatureProvider, storeInvalidBlock: ByteStr => Unit)
```

Comment:
I'd use some
```
trait InvalidBlockStorage {
def store()
def exists()
}
```
This is more obvious when you read/support/further develop the functionality
It's more consistent  with existing  repositories like utx, peerdatabase, microblockOwners, etc